httpcore/_sync/http11.py

```diff
@@ -84,10 +85,21 @@ def handle_request(self, request: Request) -> Response:
 
         try:
             kwargs = {"request": request}
-            with Trace("send_request_headers", logger, request, kwargs) as trace:
-                self._send_request_headers(**kwargs)
-            with Trace("send_request_body", logger, request, kwargs) as trace:
-                self._send_request_body(**kwargs)
+            try:
+                with Trace(
+                    "send_request_headers", logger, request, kwargs
+                ) as trace:
+                    self._send_request_headers(**kwargs)
+                with Trace("send_request_body", logger, request, kwargs) as trace:
+                    self._send_request_body(**kwargs)
+            except WriteError:
+                # If we get a write error while we're writing the request,
```

Comment:
What happens if server just closes the request, but don't answer (I saw it in some proxies)?

Reply:
The connection is closed and the client raises an exception...

```python
httpcore.RemoteProtocolError("Server disconnected without sending a response.")
```

I've added a test case to demonstrate this.